### DISCUSSION TITLE
Need help with fetching data from a base collection using python and requests.

### DISCUSSION THREAD

**Discussion by devloper16** [pocketbase/pocketbase#4472] [Q&A]
I am using python and requests to fetch data from a collection here is my code. I problem I am having is it is only returning 5 records despite of collection having over 1000 of matching records. 

My question is, How do I define the number of records?

```
filter_expression = 'source = "HomeMade" && user = "userId"'
getrecords = session.get(
    POCKETBASE_BASE_URL + "/api/collections/Orbit/records",
    headers={"Authorization": f"Bearer {token}"},params={"filter": filter_expression}).json()
```

Thanks for help in advance,

**Comment by ganigeorgiev** [comment]
Are you sure that that there are more than 5 records matching your filter (including the List/search API rule if you have any)?

Are you sure that you are not confusing the number of json props in the returned result with the actual number of records? Note that the endpoint returns a paginated result (with a default `perPage=30`) and the records are returned under the `items` key, alongside `page`, `perPage`, `totalItems`, `totalPages`.

For more details about the list/search web API please refer to https://pocketbase.io/docs/api-records/#listsearch-records

**Reply by devloper16** [reply]
Yes indeed, for the matching filter parameter there are total of 1237 records in the collection. I even tried it with without and filters and still got only 5 records back.

Yes it tried adding page and perPage but got the same result.

```
    getrecords = session.get(
    POCKETBASE_BASE_URL + "/api/collections/Orbit/records",
    headers={"Authorization": f"Bearer {token}"},params={"page": 1, "perPage": 35, "filter": filter_expression}).json()
```

**Reply by ganigeorgiev** [reply]
> for the matching filter parameter there are total of 1237 records in the collection

Make sure that you are taking the List API rule of the collection also in consideration.

The easiest way to rule out PocketBase issue is to send your request with curl, Postman, Insomnia, etc.

Other than this, I don't know how to help you further.

**Reply by devloper16** [reply]
I'll give it a shot thank you!

**Reply by devloper16** [reply]
I getting this data through Thunder Client in vscode and it is working I am getting expected data. I also tried js sdk and it is also working so I think i'll move the data fetching function to js.

Thanks for your help.
